adserver/tests/test_tasks.py

```diff
@@ -626,3 +628,47 @@ def test_daily_update_placements(self):
         self.assertEqual(pi2_ad2.offers, 2)
         self.assertEqual(pi2_ad2.views, 2)
         self.assertEqual(pi2_ad2.clicks, 0)
+
+    def test_remove_old_report_data(self):
+        # Add a very old offer
+        old_date = timezone.now() - datetime.timedelta(days=370)
+        get(
+            Offer,
+            advertisement=self.ad1,
+            publisher=self.publisher,
+            country="CA",
+            viewed=True,
+            view_time=6,
+            clicked=True,
+            keywords=["backend"],
+            div_id="id_1",
+            ad_type_slug=self.text_ad_type.slug,
+            date=old_date,
+        )
+
+        # Run aggregations both for today and the very old day
+        update_previous_day_reports(timezone.now())
+        update_previous_day_reports(old_date)
+
+        # Check that the aggregations match
+        impression_old = RegionTopicImpression.objects.filter(
+            region="us-ca",
+            topic="backend-web",
+            advertisement=self.ad1,
+            date=old_date.date(),
+        ).first()
+        self.assertIsNotNone(impression_old)
+        self.assertEqual(impression_old.offers, 1)
+        self.assertEqual(impression_old.views, 1)
+        self.assertEqual(impression_old.clicks, 1)
+
+        # Remove old aggregation data and verify they are gone
+        remove_old_report_data()
+        self.assertFalse(
+            RegionTopicImpression.objects.filter(
+                region="us-ca",
+                topic="backend-web",
+                advertisement=self.ad1,
+                date=old_date.date(),
+            ).exists()
+        )
```

Comment:
I wonder if we should also have a newer Offer, and ensure that one doesn't get deleted?

Reply:
There is a newer one so probably just need a separate check to ensure it isn't deleted.

Reply:
I added a test for it.